src/main/java/ortus/boxlang/runtime/dynamic/casters/DateTimeCaster.java

```diff
@@ -147,7 +163,7 @@ public static DateTime cast( Object object, Boolean fail, ZoneId timezone ) {
 
 		// We have a DateTime object
 		if ( object instanceof DateTime targetDateTime ) {
-			return targetDateTime;
+			return clone ? targetDateTime.clone() : targetDateTime;
```

Comment:
I am not a fan of doing this in the caster at all.   I think it's OK to do an instance check in the `DateAdd` BIF and then clone it if it's a `DateTime` object.  This is going to create clones of every reference used and passed in everywhere.